extensions/resource-deployment/src/test/services/platformService.test.ts

```diff
@@ -0,0 +1,343 @@
+					sinon.stub(vscode.workspace, 'textDocuments').get(() => [{ isUntitled: true, fileName: '' }]);
+				}
+				const result = platformService.isNotebookNameUsed(title);
+				result.should.equal(isUsed);
+			});
+		}));
+	});
+	describe('runCommand', () => {
+		[
+			//[commandSucceeds, ignoreError]
+			[true, undefined],
+			[false, true],
+			[false, false],
+		].forEach(([commandSucceeds, ignoreError]) => {
+			if (ignoreError && commandSucceeds) {
+				return; //exit out of the loop as we do not handle ignoreError when command is successful
+			}
+			it(`non-sudo, commandSucceeds: ${commandSucceeds}, ignoreError: ${ignoreError}`, async () => {
+				const command = __dirname; // arbitrary command string, and success string on successful execution and error string on error
+				const child = new TestChildProcessPromise<cp.Output>();
+				const stub = sinon.stub(cp, 'spawn').returns(child);
+				const runningCommand = platformService.runCommand(command, { commandTitle: 'title', ignoreError: ignoreError });
+				// fake runCommand to behave like echo, returning the command back as stdout/stderr/error.
+				// TestChildProcessPromise object shares the stdout/stderr stream for convenience with the child stream.
+				if (commandSucceeds) {
+					child.emit('data', command);
+					child.emit('exit', 0, null); //resolve with 0 exit code
+					child.resolve({ stdout: command });
+				} else {
+					child.emit('data', command);
+					child.emit('exit', 1, null); // resolve with non-zero exit code
+					child.reject({ stderr: command });
+				}
+				const { result, error } = await tryExecuteAction(() => runningCommand);
+				verifyCommandExecution(stub, result, error, command, commandSucceeds, ignoreError);
+			});
+
+			it(`sudo, commandSucceeds: ${commandSucceeds}, ignoreError: ${ignoreError}`, async () => {
+				const command = __dirname; // arbitrary command string, and success string on successful execution
+				const stub = sinon.stub(sudo, 'exec').callsFake((cmd, _options, cb) => {
+					// behaves like echo, returning the _cmd back as stdout/stderr/error.
+					if (commandSucceeds) {
+						cb(''/* error */, cmd/* stdout */, ''/* stderr */);
+					} else {
+						cb(cmd/* error */, ''/* stdout */, cmd/* stderr */);
+					}
+				});
+				const { error, result } = await tryExecuteAction(() => platformService.runCommand(command, { commandTitle: 'title', ignoreError: ignoreError, sudo: true, workingDirectory: __dirname }));
+				verifyCommandExecution(stub, result, error, command, commandSucceeds, ignoreError);
+			});
+		});
+	});
+});
+
+function verifyCommandExecution(stub: sinon.SinonStub, result: string | undefined, error: any, command: string, commandSucceeds: boolean | undefined, ignoreError: boolean | undefined) {
+	stub.callCount.should.equal(1);
+	if (commandSucceeds) {
+		result!.should.equal(command);
+	} else {
+		if (ignoreError) {
+			should(error).be.undefined();
+			(result === undefined || result.length === 0).should.be.true();
```

Comment:
These should have messages saying what is failing - it's not going to display anything useful just by checking true/false

Reply:
Added message.